src/component.jl

```diff
@@ -0,0 +1,385 @@
+"""
+function get_forecast(
+                      ::Type{T},
+                      component::InfrastructureSystemsType,
+                      initial_time::Dates.DateTime,
+                      label::AbstractString,
+                      horizon::Int,
+                     ) where T <: Forecast
+    if !has_forecasts(component)
+        throw(ArgumentError("no forecasts are stored in $component"))
+    end
+
+    first_forecast = iterate(iterate_forecasts(component))[1]
+    resolution = get_resolution(first_forecast)
+    sys_horizon = get_horizon(first_forecast)
+
+    forecast = get_forecast(
+        forecast_external_to_internal(T),
+        component,
+        initial_time,
+        resolution,
+        sys_horizon,
+        label,
+        horizon,
+    )
+
+    index = Int((initial_time - get_initial_time(forecast)) / resolution) + 1
+    ts = get_time_series(
+        _get_time_series_storage(component),
+        get_time_series_uuid(forecast),
+        get_name(component);
+        index=index,
+        len=horizon,
+    )
+
+    return make_public_forecast(forecast, ts)
+end
+
+function get_forecast(
+                      ::Type{T},
+                      component::InfrastructureSystemsType,
+                      initial_time::Dates.DateTime,
+                      label::AbstractString,
+                     ) where T <: ForecastInternal
+    return get_forecast(T, _get_forecast_container(component), initial_time, label)
+end
+
+function get_forecast(
+                      ::Type{T},
+                      component::InfrastructureSystemsType,
+                      initial_time::Dates.DateTime,
+                      sys_resolution::Dates.Period,
+                      sys_horizon::Int,
+                      label::AbstractString,
+                      horizon::Int,
+                     ) where T <: ForecastInternal
+    end_time = initial_time + sys_resolution * horizon
+    @debug "Requested forecast" initial_time horizon
+    initial_times = get_forecast_initial_times(T, _get_forecast_container(component), label)
+    for it in initial_times
+        # Return the forecast for this time array if it encompasses the requested range and
+        # one of its data points is the requested initial_time.
+        if it <= initial_time && (end_time <= (it + sys_resolution * sys_horizon)) &&
+                ((initial_time - it) % Dates.Millisecond(sys_resolution) == Dates.Second(0))
+            return get_forecast(T, _get_forecast_container(component), it, label)
+        end
+    end
+
+    throw(ArgumentError("did not find a forecast matching the requested parameters"))
+end
+
+"""
+    get_forecast_values(component::InfrastructureSystemsType, forecast::Forecast)
+
+Return a TimeSeries.TimeArray where the forecast data has been multiplied by the forecasted
+component field.
+"""
+function get_forecast_values(component::InfrastructureSystemsType, forecast::Forecast)
```

Comment:
@jd-lara Can you confirm that this is what you want?

Reply:
No, this is what I am looking for 

```Julia
function get_forecast_values(component::InfrastructureSystemsType, label)
```

Reply:
If you don't want to pass the forecast object then it would have to be
```
function get_forecast_values(::Type{T<:Forecast}, component::InfrastructureSystemsType, initial_time, label)
```